pkg/daemon/config_drift_monitor_test.go

```diff
@@ -0,0 +1,414 @@
+package daemon
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	ign3types "github.com/coreos/ignition/v2/config/v3_2/types"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/openshift/machine-config-operator/test/helpers"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+func TestConfigDriftMonitor(t *testing.T) {
+	// Our errors. Their contents don't matter because we're not basing our
+	// assertions off of their contents. Instead, we're more concerned about
+	// their types.
+	fileErr := configDriftErr(fileConfigDriftErr(fmt.Errorf("file error")))
+	unitErr := configDriftErr(unitConfigDriftErr(fmt.Errorf("unit error")))
+
+	// Filesystem Mutators
+	// These don't have to be closures. I just wanted to avoid potential
+	// namespace collisions.
+	changeFileContent := func(path string) error {
+		return ioutil.WriteFile(path, []byte("notthecontents"), defaultFilePermissions)
+	}
+
+	touchFile := func(path string) error {
+		now := time.Now()
+		return os.Chtimes(path, now.Local(), now.Local())
+	}
+
+	renameFile := func(path string) error {
+		return os.Rename(path, path+".old")
+	}
+
+	overwriteFile := func(path string) error {
+		tmpPath := filepath.Join(filepath.Dir(path), "overwritten")
+		if err := writeFileAtomicallyWithDefaults(tmpPath, []byte("notthecontents")); err != nil {
+			return err
+		}
+
+		return os.Rename(tmpPath, path)
+
+	}
+
+	chmodFile := func(path string) error {
```

Comment:
Is this function for file or directory? Directory usually have permission 755 but a file would have 644

Reply:
It's for a file. The test uses `defaultFilePermission` when it creates the file. This function just modifies the file permissions to something that is not 644 to cause the Config Drift Monitor to report that a config drift has occurred.

Reply:
aha, interesting. Makes sense